Core/clim-core/panes.lisp

```diff
@@ -363,14 +363,23 @@ order to produce a double-click")
   (print-unreadable-object (pane sink :type t :identity t)
     (prin1 (pane-name pane) sink)))
 
+(defun coerce-pane-type (type)
+  (cond ((not (keywordp type))
+         type)
+        ((let ((name (symbol-name type))
+               (package (load-time-value (find-package '#:clim))))
+           (or (find-symbol name package)
+               (let* ((suffix (load-time-value (string '#:-pane)))
+                      (name+pane (concatenate 'string name suffix)))
+                 (find-symbol name+pane package)))))
+        (t
+         type)))
+
 (defun make-pane (type &rest args)
-  (when (eql (symbol-package type)
-             (symbol-package :foo))
-    (setf type (or (find-symbol (symbol-name type) (find-package :clim))
-                   type)))
-  (apply #'make-pane-1 (or *pane-realizer*
-			   (frame-manager *application-frame*))
-	 *application-frame* type args))
+  (let* ((type     (coerce-pane-type type))
+         (frame    *application-frame*)
+         (realizer (or *pane-realizer* (frame-manager frame))))
+    (apply #'make-pane-1 realizer frame type args)))
```

Comment:
I agree that the changes are incomplete at best and more likely plain wrong. Fixing `make-pane` and the duplication in the backends is of course out of scope.

> Generate clause genearate-ensure-pane-form in your code should invoke make-pane-1 with the appropriate (see make-pane) realizer without trying to coerce the type on its own.

The problem is that the "ensure pane" logic needs to know the concrete pane type to decide whether to reinitialize or throw away an existing pane.

The proper solution would probably be a protocol for resolving abstract pane types that dispatches on the realizer so the "ensure pane" logic could just ask for the concrete pane type instead of half-assedly trying to compute it itself.

Reply:
I've submitted a PR which does "the right thing", that is takes into account that a pane-realizer may substitute a class in CLIM package with its own.